core/blockchain.go

```diff
@@ -1669,8 +1673,8 @@ func (bc *BlockChain) writeKnownBlock(block *types.Block) error {
 
 // WriteBlockWithState writes the block and all associated state to the database.
 func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.Receipt, logs []*types.Log, state *state.StateDB, emitHeadEvent bool) (status WriteStatus, err error) {
-	bc.chainmu.Lock()
-	defer bc.chainmu.Unlock()
+	bc.chainLock.LockLow()
```

Comment:
Why use `LockLow` here? I think `WriteBlockWithState` is on the critical path.

Reply:
yes, you are right